### DISCUSSION TITLE
[@lit-labs/react] createComponent with generic types

### DISCUSSION THREAD

**Discussion by muratcorlu** [lit/lit#3700] [Q&A]
I have a custom Lit select component that uses Generic types for its `value` attribute and custom events.

```ts
@customElement('bl-select')
export default class BlSelect<ValueType extends FormValue = string> extends FormControlMixin(LitElement) {
```

I couldn't manage to be able to pass this generic type from React.

My Current situation:

```ts
export const BlSelect = React.lazy<ReactWebComponent<BlSelectType, {
  onBlSelect: EventName<CustomEvent<ISelectOption<ValueType>[]>>
}>>(() =>
  customElements.whenDefined('bl-select').then(elem => ({
      default: createComponent<BlSelectType>(
        {
          react: React,
          tagName: 'bl-select',
          elementClass: elem,
          events: {"onBlSelect":"bl-select"}
        }
      )
    })
));
```

Of course, `ValueType` in code above doesn't mean anything. I need to get that type from the user of `BlSelect` react component.

Any ideas how to manage this?

**Comment by Haprog** [comment]
I noticed your question on StackOverflow and answered there. Please see my answer at https://stackoverflow.com/a/77889381/1362634

My solution feels like a bit of a hack or workaround but it seems to work well enough to be able to use generics from React for the wrapped web component.
